src/graph/impl/KokkosGraph_Distance1Color_impl.hpp

```diff
@@ -2293,63 +2250,32 @@ class GraphColor_VB:public GraphColor <HandleType,lno_row_view_t_,lno_nnz_view_t
   };
 
 
-  /**
-   * Functor for parallel prefix sum
-   */
-  template <typename view_type>
-  struct parallel_prefix_sum{
-    view_type _vertexList;
-    view_type _recolorList;
-    view_type _pps_view;
-
-    parallel_prefix_sum(
-        view_type vertexList,
-        view_type recolorList,
-        view_type pps_view):
-          _vertexList(vertexList),_recolorList(recolorList),_pps_view(pps_view){}
-
-    KOKKOS_INLINE_FUNCTION
-    void operator()(const typename view_type::non_const_value_type ii, size_t& update, const bool final) const {
-      typename view_type::non_const_value_type w = _vertexList(ii);
-      update += _recolorList(w);
-      if (final) {
-        _pps_view(w) = (update);
-      }
-    }
-  };
-
-
-  /**
-   * Functor for creating new worklist using pps
-   */
   template <typename view_type>
-  struct create_new_work_array{
+  struct ppsWorklistFunctorVB {
```

Comment:
Is the idea we are replacing generic PPS with something that works mainly for VB ?